src/winforms/toga_winforms/widgets/base.py

```diff
@@ -26,6 +26,7 @@ def container(self, container):
         self._container = container
         if self.native:
             self._container.native.Controls.Add(self.native)
+            self.native.BringToFront()
```

Comment:
Forgive my ignorance -- I have very little experience with Winforms -- can you explain why `BringToFront` is necessary when adding the widget to the container? What would otherwise cover it? I'm just curious because other several other platforms (GTK, Cocoa) aren't doing this.

Reply:
I have to research this further, I really don't know why the widgets were not brought forward automatically. 
I've actually spent two days working on the table example and box implementation because when I ran the example after adding the native for toga `Box` widget,  only the top part of the `label` was visible at the bottom of the window. I've tried to find when the `set_hidden` was called on widgets, but it wasn't, and the tables were still not visible. So I found this solution on stackoverflow, but it doesn't seem to be necessary, as you say.

Reply:
If I had to guess, it will be about insertion order. If widgets are placed in z-index order matching insertion order, adding a parent and then a child will result in the child being "behind" the parent. That doesn't make a whole lot of sense for the sort of layouts that *we* are doing, but it's just as valid as a design decision - and it's easily reversed by bringing every widget to the front after it has been added.

Reply:
Your guess is correct: z-index depends on the order the controls (widgets) are added.  At least that's what I've read in "Pro .NET 2.0 Windows Forms and Custom Controls in VB 2005" on google books.
So, bringing each child to front is OK, isn't it?

Reply:
Absolutely - if only because "it works" :-)

Longer term, if there's a way to reverse the order of insertion (i.e., make the widget tree traversal "add children, then add parent", rather than "add parent, then add children"), that should remove the need for the extra call - but I'm Ok with using `BringToFront()` in the meantime.